api/swagger_v2.yaml

```diff
@@ -0,0 +1,1138 @@
+    subscribing compute nodes. HMNFD provides the ability to notify subscribers
+    of component hardware state changes and other changes made to and by the
+    Hardware State Manager.
+
+    To receive notifications, a compute node must have an http or https based HMNFD API
+    endpoint service running. This is where the State Change Notifications will
+    be sent when they occur. HMNFD works with the Hardware State Manager and
+    distributes state changes and manages subscriptions.
+
+
+    The REST API provides the following functions:
+
+    * Subscription for component state changes:
+        * Hardware state (On, Off, Ready, Standby, Halt etc.)
+        * Logical state (arbitrary like AdminDown, Other)
+        * Role (Compute, Management, Application etc.)
+        * Enabled state
+        * Flag (OK, Alert, Warning, etc.)
+
+    * View current subscriptions
+
+    * Delete subscriptions
+
+    * Retrieve and modify current service operating parameters
+
+    * Create state change notifications for distribution to subscribers
+
+    ## Deprecation Notice: V1 of the HMS service has been deprecated as of CSM version 1.2.0. 
+    The V1 HSM API’s will be removed in the CSM version 1.6 release.
+    All consumers of the V1 HSM API interface will need to move to the V2 interface prior to the CSM 1.6 release.
+
+    ## Resources
+
+    ### /subscriptions
+
+    Manage and view subscriptions to notifications. This resource is generally 
+    used by compute nodes.
+
+    ### /scn
+
+    State change notification messages sent from Hardware State Manager to
+    HMNFD; the same format is used for notifications sent by HMNFD to
+    subscribers. This resource applies only to the HSM.
+
+    ### /params
+
+    Retrieve or update configurable parameters.
+
+    ### /health
+
+    Retrieve health information for the service and its dependencies.
+
+    ## Workflows
+
+    ### Manage SCN Subscriptions
+
+    #### GET /subscriptions
+
+    Retrieve and view current subscriptions.
+
+    #### POST /subscriptions/{xname}/agents/{agent}
+
+    A node will subscribe to whatever state change notifications (SCNs) it wants
+    to receive. Thus, the node needs to have a service running to which HMNFD can
+    send, via REST, the subscribed-to SCNs. A URL that tells where to send the
+    SCNs is provided as part of the request body schema. Once a subscribed-to
+    SCN occurs, NFD sends it to the node's service via a REST call to the URL
+    supplied during the subscribe operation.  The {xname} and {agent} segments
+    of the URL state which node and which agent on that node are doing the
+    subscribing.  The subscription request info is in the body of the PUT 
+    payload.
+
+    #### PATCH /subscriptions/{xname}/agents/{agent}
+
+    Modify an existing SCN subscription.  Component list, component states, etc.
+    can be modified with this request, which will remain associated with the
+    component and software agent specified in the URL.
```

Comment:
I think this agent definition is good and probably need to be repeated in the description of the PATCH/ POST/ DELETE

Reply:
Fixed.